### PR TITLE
fix: Limit p2p protocol message size

### PR DESCRIPTION
**Pull request by TheWaWaR**

### CODE REVIEW

**Comment by driftluo** [comment]
`send_bulk_of_tx_hashes` must be pruned when too large

File: sync/src/relayer/get_transactions_process.rs [hunk 1]
```diff
@@ -65,6 +67,15 @@ impl<'a> GetTransactionsProcess<'a> {
                         .transaction(tx.data())
                         .build()
                 })
+                .take_while(|relay_tx| {
+                    // If the size of RelayTransactions response is too large we
+                    // just skip it, since this case is rare and we can get
+                    // missing transactions by BlockTransactions message later.
+                    total_size += relay_tx.as_slice().len();
```
Comment (doitian) [inline sync/src/relayer/get_transactions_process.rs]: may overflow

**Comment by quake** [comment]
I am working on it, will create a new pr later

File: sync/src/relayer/get_transactions_process.rs (hunk 1 above)
Comment (TheWaWaR) [reply sync/src/relayer/get_transactions_process.rs]: Actually it's not possible, total_size must not greater than `MAX_RELAY_TXS_SIZE`

`total_size <= MAX_RELAY_TXS_SIZE`

**Comment by doitian** [comment]
bors r=quake,doitian,driftluo,u2

**Pull request merged**